### ISSUE TITLE
User.get_name() breaks AD authentication

### ISSUE THREAD

**Issue by nchowning** [shinken-monitoring/mod-webui#501]
Yesterday we noticed that contacts without an email address were acting very flakey when attempting to login.

```
define contact {
        contact_name                   downtime_user
        alias                          shinken downtime user
        use                            generic-admin
}
```

Looking at the brokerd log, I noticed that sometimes `Active Directory UI` was using the contact name s the samaccountname and sometimes it was using the contact alias.

```
[1462810001] INFO: [broker-master] [WebUI] Authenticating user 'downtime_user' with ActiveDir_UI
[1462810001] INFO: [broker-master] [Active Directory UI] AD/Ldap Connection done
[1462810001] INFO: [broker-master] [Active Directory UI] AD/LDAP: search for contact downtime_user
[1462810001] INFO: [broker-master] [Active Directory UI] Filter (| (samaccountname=downtime_user)(mail=none))
[1462810001] INFO: [broker-master] [Active Directory UI] Find account principalname downtime_user@redacted
[1462810001] INFO: [broker-master] [Active Directory UI] AD/Ldap Connection done with user downtime_user
```

```
[1462815427] INFO: [broker-master] [WebUI] Authenticating user 'downtime_user' with ActiveDir_UI
[1462815427] INFO: [broker-master] [Active Directory UI] AD/Ldap Connection done
[1462815427] INFO: [broker-master] [Active Directory UI] AD/LDAP: search for contact shinken downtime user
[1462815427] INFO: [broker-master] [Active Directory UI] Filter (| (samaccountname=shinken downtime user)(mail=none))
[1462815427] WARNING: [broker-master] [Active Directory UI] No result for shinken downtime user
```

This is not a problem for contacts with an email address since the Active Directory UI Filtering includes the email address. After some sleuthing, I believe I've tracked down the issue:

Initially `auth-active-directory.AD-Webui.find_contact_entry` receives a contact in the form of `shinken.objects.contact.Contact`. `shinken.objects.contact.Contact.get_name` return's the contact's username - in this case: `downtime_user`.

Other times, `auth-active-directory.AD-Webui.find_contact_entry` receives a contact in the form of `webui2.user.User`. `webui2.user.User.get_name` has logic to return the contact name, realname, or alias:

```
def get_name(self):
    name = self.get_username()
    if getattr(self, 'realname', None):
        name = "%s %s" % (getattr(self, 'firstname'), getattr(self, 'realname'))
    elif getattr(self, 'alias', None) and getattr(self, 'alias', None) != 'none':
        name = getattr(self, 'alias', name)
    return name
```

It would appear that `webui2.user.User.get_username` is the functionality that is needed for `auth-active-directory.AD-Webui.find_contact_entry`. I'm trying to determine the best way to resolve this issue.

* Hack on `auth-active-directory.AD-Webui.find_contact_entry` to add a try/catch to try to use `get_username()` and use `get_name()` otherwise. This method works.
* Change `webui2.user.User.get_name` to behave more like `shinken.objects.contact.Contact` and return the contact username. This would likely have other implications but I'm not sure how wide reaching they would be.

**Comment by mohierf** [comment]
Are you using the most recent version of WebUI ?

Anyway, the WebUI is calling the `check_auth` function of the module and provides two strings : `username` and `password` (see: https://github.com/shinken-monitoring/mod-auth-active-directory/blob/master/module/module.py#L213). Some few lines later (https://github.com/shinken-monitoring/mod-auth-active-directory/blob/master/module/module.py#L220), it instantiates a `Contact` from the Shinken datamanager and then finally provides this object to the `find_contact_entry` function ... **but** you can not use any WebUI `User` object function in this source code because this module uses the Shinken Datamanager and not the WebUI one :disappointed: 

 IMHO, you should hack the module near https://github.com/shinken-monitoring/mod-auth-active-directory/blob/master/module/module.py#L136 to  make it use the `contact_name` property ... as far as I understood clearly your problem?

 Anyway, this problem is not an issue in the WebUI, but rather in the AD module ... please note that the last version ot this module is published for Shinken version 1.4!

**Comment by mohierf** [comment]
No news. Closing
